### ISSUE TITLE
Stopping the hide timeout via `update` functions

### ISSUE THREAD

**Issue by silverwind** [almond-bongbong/react-simple-toasts#93]
Use case: Show toast with a duration but stop the hide timeout when the user mouses into the tooltip.

In my `onMouseEnter`, I call `update({duration: 0})` which ends up [here](https://github.com/almond-bongbong/react-simple-toasts/blob/dc65ffaf4844083b06a19b0828f1ae7c6eb32d2a/src/index.tsx#L209-L210), but values `0` or `null` are being rejected by the function so `clearTimeout` is never called.

I think as a fix, the function should gain a boolean parameter to know whether it's been called from `update` and accept `null` and `0` to trigger the `clearTimeout`.

**Comment by almond-bongbong** [comment]
Thank you so much for your suggestion! The feature you proposed seems like it would be very helpful for other users as well.

In fact, it's such a useful idea that we might consider making it a built-in option to pause the timer when a user hovers over the toast. This would make it even easier to use.

Regarding your suggested implementation of using 0, null, or Infinity values in the update function, I think we may need to give that some more thought. It might be clearer to provide separate stop and start methods instead.

Overall, I'm leaning towards creating a built-in option that pauses the timer when the user hovers over the toast. What do you think about this approach?

**Comment by silverwind** [comment]
Yes, pausing the timer (as opposed to completely stopping, which I was initially suggesting) sounds like the best approach, and I agree it could be enabled by default.

> Regarding your suggested implementation of using 0, null, or Infinity values in the update function, I think we may need to give that some more thought. It might be clearer to provide separate stop and start methods instead.

I think it's clear that when those values are passed during `update` that the user wants to alter the timeout and all three values mean "no timeout", so I would stop it.
